### ISSUE TITLE
Application is crashing when some component calls nanovgoSurface.Stroke()

### ISSUE THREAD

**Issue by lukebayes** [lukebayes/epiphyte#127]

**Comment by lukebayes** [comment]
If I comment out the nascent FPS control, or comment out the calls to Stroke() in views.go, the crash doesn't happen.

It feels like there's a dependent call that's missing, but I'm not sure what it is yet.

Added support for BeginPath() to the surface interface, and then ensured we called it in the same places that Shibukawa seems to (e.g., before DrawRectangle(), etc.), but that didn't fix it.

Any thoughts would be welcomed.

```
make run
/Users/lukebayes/Projects/epiphyte/lib/go-1.10/bin/go run ./src/examples/boxes/main.go
panic:

goroutine 1 [running, locked to thread]:
github.com/shibukawa/nanovgo.(*Context).Stroke(0xc4200dc000)
        /Users/lukebayes/Projects/epiphyte/vendor/src/github.com/shibukawa/nanovgo/nanovgo.go:757 +0x470
display.(*nanoSurface).Stroke(0xc42000e0e0)
        /Users/lukebayes/Projects/epiphyte/src/display/nano_surface.go:30 +0x2e
display.(*OffsetSurface).Stroke(0xc42000cbe0)
        /Users/lukebayes/Projects/epiphyte/src/display/offset_surface.go:55 +0x33
display.RectangleView(0x4240660, 0xc42000cbe0, 0x42412c0, 0xc4200be3f0, 0x413addb, 0x42412c0)
        /Users/lukebayes/Projects/epiphyte/src/display/views.go:11 +0x1b2
display.LabelView(0x4240660, 0xc42000cbe0, 0x42412c0, 0xc4200be3f0, 0xc42000cbe0, 0xc420067d60)
        /Users/lukebayes/Projects/epiphyte/src/display/views.go:16 +0x50
display.(*Component).Draw(0xc4200be3f0, 0x42406e0, 0xc42000e0e0)
        /Users/lukebayes/Projects/epiphyte/src/display/component.go:735 +0x88
display.(*Component).DrawChildren(0xc4200be360, 0x42406e0, 0xc42000e0e0)
        /Users/lukebayes/Projects/epiphyte/src/display/component.go:729 +0x73
display.(*Component).Draw(0xc4200be360, 0x42406e0, 0xc42000e0e0)
        /Users/lukebayes/Projects/epiphyte/src/display/component.go:736 +0xaa
display.(*Component).DrawChildren(0xc4200965b0, 0x42406e0, 0xc42000e0e0)
        /Users/lukebayes/Projects/epiphyte/src/display/component.go:729 +0x73
display.(*Component).Draw(0xc4200965b0, 0x42406e0, 0xc42000e0e0)
        /Users/lukebayes/Projects/epiphyte/src/display/component.go:736 +0xaa
display.(*NanoWindowComponent).LayoutDrawAndPaint(0xc4200965b0)
        /Users/lukebayes/Projects/epiphyte/src/display/nano_window.go:112 +0x17e
display.(*NanoWindowComponent).Init(0xc4200965b0)
        /Users/lukebayes/Projects/epiphyte/src/display/nano_window.go:81 +0x1ba
main.main()
        /Users/lukebayes/Projects/epiphyte/src/examples/boxes/main.go:76 +0x6b
```

**Comment by lukebayes** [comment]
Fixed in PR #129 

It looks like the nanovgo wrapper caches vertex paths and panics if Stroke() is called when that cache has more than zero entries.

It's possible this was related to calling BeginPath() before Rect() and then trying to call Fill() & Stroke() without calling BeginPath() and Rect() a second time.

I did try this before, but it also looks like the second call to Rect() has to have different values than the first one in order to avoid populating the cache.

Not 100% sure this is really the problem or the solution, but the crash has stopped happening now.
